lib/tasks_private/instance_types.rake

```diff
@@ -1,215 +1,277 @@
-    instance[:disabled] = true
-    output_instances << instance
-  end
+    def http_get(uri, headers = {})
+      LOGGER.info("Making request to #{uri}")
 
-  # new discontinued from current_deprecated
-  discontinued_count += current_deprecated.count
-  current_deprecated.values.each do |instance|
-    instance[:disabled] = true
-    output_instances << instance
-  end
+      response = Net::HTTP.start(uri.host, :use_ssl => true) do |http|
+        http.get(uri.request_uri, headers)
+      end
 
-  new_instance_types_file << output_instances.map { |i| format_instance(i) }.join("\n")
-  new_instance_types_file << <<EOT
-  }.freeze
+      body = response.body if response.code_type.body_permitted?
+      json = response['content-type'] =~ JSON_CONTENT_RE || uri.path.end_with?('json')
+      data = JSON.parse(body) if json && body.present?
 
-  def self.instance_types
-    additional = Hash(Settings.ems.ems_amazon.try!(:additional_instance_types)).stringify_keys
-    disabled = Array(Settings.ems.ems_amazon.try!(:disabled_instance_types))
-    AVAILABLE_TYPES.merge(DEPRECATED_TYPES).merge(DISCONTINUED_TYPES).merge(additional).except(*disabled)
-  end
+      unless response.kind_of?(Net::HTTPSuccess)
+        LOGGER.error(data['message']) if data&.key?('message')
+        LOGGER.error("Error getting data from #{uri}")
+        response.error!
+      end
 
-  def self.all
-    instance_types.values
-  end
+      data.presence || body.presence
+    end
 
-  def self.names
-    instance_types.keys
-  end
-end
-EOT
+    def get_gh_data(uri_path, query_data = {})
+      uri = URI::HTTPS.build(
+        :host  => GH_API_HOST,
+        :path  => "/repos/#{SDK_REPO_PATH}/#{uri_path}",
+        :query => (query_data.to_query unless query_data.empty?),
+      )
+      http_get(uri, GH_API_HEADERS)
+    end
 
-  File.open(ManageIQ::Providers::Amazon::Engine.root.join('instance_types.rb'), 'w') do |f|
-    f.puts(new_instance_types_file)
-  end
+    def get_gh_file(file_path, sha: nil)
+      if sha.blank?
+        commits = get_gh_data('commits', :path => file_path, :page => 1, :per_page => 1)
+        sha = commits[0]['sha']
+        LOGGER.debug("Latest commit SHA of '#{file_path}' file is #{sha}")
+      end
+
+      GH_CACHE.fetch("#{file_path}.#{sha}".tr('/', '.')) do
+        path = "/#{SDK_REPO_PATH}/#{sha}/#{file_path}"
+        http_get(URI::HTTPS.build(:host => GH_RAW_HOST, :path => path))
+      end
+    end
+
+    discontinued_types, types_list = isolated do
+      previous_list = []
+      versions = get_gh_data("contents/#{SDK_DATA_DIR}")
+      versions = versions.lazy.select { |v| v['type'] == 'dir' }.map { |v| v['name'] }.sort
+      all_types, latest = versions.reduce([[], []]) do |(all, _latest), version|
+        data = get_gh_file("#{SDK_DATA_DIR}/#{version}/#{SDK_FILE_NAME}")
+        data = data['shapes']['InstanceType']['enum']
+        old_minus_new = previous_list - data
+        new_minus_old = data - previous_list
```

Comment:
removed/added might be better variable names for readability.

Reply:
is that critical?

Reply:
Others would say no, but to me readability and maintenance are crticial. :smile:  old_minus_new says nothing about the intent, which causes more cognitive overhead, as compared to a name like `removed`